### DISCUSSION TITLE
v20.0.0 First Stable Release!

### DISCUSSION THREAD

**Discussion by jonyo** [geodesicsolutions-community/geocore-community#181] [Announcements]
This is the first non-beta release of the **Community Edition**!  There is more work to be done, just take a look at the open issues, but it has been tested as much as possible and all of the main functionality is in place and working.

# Stability

The community is growing but small at the moment, so "first stable release" is relative.  We've tested everything we can think of but there may be some features you used in the past that may or may not work.

As always, **test before using** on a production website.

# Compatibility

This version works on PHP 7.4 or higher, and PHP 8.0 and 8.1.  In fact we did most of our testing on PHP 8.1.

## What's Changed - all changes since v18.02 on legacy platform

Note: these include all changes in all the 20.0.0 beta versions, so it includes everything done since the last "commercial" release of the software by the now closed company (since version 18.02 released commercially). 

* Import legacy code - this imports the code with a fresh start from latest release by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/7
* Remove log license addon by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/8
* Apply all the un-released changes from legacy codeset by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/10
* Remove OS files and add them to .gitignore by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/11
* Update installation requirements check and links by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/14
* Add docker support by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/17
* Update the admin panel wiki links to go to the new URL by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/19
* Fleshing out the docker instructions by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/20
* Add docker install link by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/21
* Fix the error on the cron admin page by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/23
* Getting ready to make a release package by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/25
* Fix several PHP deprecation notices in the admin panel. by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/30
* Fixing build to include user_images, also seperating extra template sets by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/35
* Update version number to 20.0.0 in software by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/38
* Add a link to the license agreement on github on install and upgrade pages by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/39
* Update old copyright notices to say distributed freely under MIT license by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/40
* Add link to github releases in the list of changelogs by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/43
* Fix the PHP version it shows when upgrading and the version check fai… by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/46
* Admin login re-skin for GeoCore CE by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/47
* Make links during setup include the port to work better in local dock… by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/48
* Fix SQL query error in storefront cron, trying to pass a query param … by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/51
* Going through results of cs-check and cleaning up a bunch of files by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/52
* Update ADODB to latest version, and switch to using composer by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/54
* Fix up lint issues on the previously encoded files, and remove some deprecated methods by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/69
* Vscode xdebug settings that work with docker by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/70
* Removing adploter addon since the service is now defunct. by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/73
* Add PHPStan tool to help identify bugs, and fix a few of them by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/82
* Fix PHP7 deprecation warnings on a few remaining addon files by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/84
* Cleanup - improvements to the build script by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/87
* Fix a few PHP8 compatibility problems by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/88
* Update smarty library to 4.1.0, rewrite how templates are loaded, and some PHP8 compatibility changes by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/90
* Fix some PHP8 errors in share fees addon and others by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/98
* Remove deprecated create_function calls PHP8 compatibility by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/100
* Update addon templates - some {include} needed updating by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/101
* Disable the post install script for now since pre-commit hook does no… by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/104
* Lotsa fixes - fixing slew of minor things in one PR by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/105
* Fixing function definitions to match parent class PHP 8 compatibility by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/107
* Fixing addon includes in admin panel by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/108
* Fix calls to fetch to use full addon/name path by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/109
* Fix PHP8 errors about calling function statically by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/110
* Fixed the curly brace error (PHP 7.4 compatibility) in PEAR libraries… by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/112
* Fixing up the exporter export functionality for PHP8 by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/114
* Add a new composer script to do all the scans and output to files to … by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/117
* Replaces sample banners with a generic vs real affiliate banners by @vicos59 in https://github.com/geodesicsolutions-community/geocore-community/pull/118
* Changes 'Powered by' branding to GeoCore CE by @vicos59 in https://github.com/geodesicsolutions-community/geocore-community/pull/121
* Fix problems with editing user in admin, and make sure new user field is included in new installation by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/124
* Fix addon management in PHP8 when trying to install an addon with exc… by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/125
* Fixing unset vars and divide by 0 errors on my account page by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/126
* Fix several PHP8 compatibility issues when placing a listing by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/127
* Fix layout and PHP8 issues when editing attachments and it is read only by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/128
* Update admin cart to use the full include path so it works by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/130
* Fix PHP8 compatibility issues browsing categories by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/131
* Fix PHP8 errors on admin home page in version box by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/132
* Removed obsolete Flash Embeds for YouTube videos by @vicos59 in https://github.com/geodesicsolutions-community/geocore-community/pull/129
* Create community code of conduct by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/133
* Fixing PHP 8 issues on grid view, including changes to main_page temp… by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/136
* Removed affiliate banners, Google+ from alternate template sets by @vicos59 in https://github.com/geodesicsolutions-community/geocore-community/pull/141
* Add .gitattributes to normalize line endings, and fix a bunch of PSR12 issues by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/137
* After adding .gitattributes previously, fixing crlf line endings to lf by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/144
* Remove License Info page from admin by @vicos59 in https://github.com/geodesicsolutions-community/geocore-community/pull/145
* remove legacy changelogs from upgrade folder by @vicos59 in https://github.com/geodesicsolutions-community/geocore-community/pull/146
* Strip RSS News Reader, clean up Support/Download blocks on admin home by @vicos59 in https://github.com/geodesicsolutions-community/geocore-community/pull/147
* Create missing folder src/addons/exporter/exports/ by @vicos59 in https://github.com/geodesicsolutions-community/geocore-community/pull/151
* Adjusting build to exclude any exports made by exporter addon when te… by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/154
* Fix issues with the SEO addon admin panel by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/169
* Fix the number of addons installed/enabled that is displayed on admin… by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/170
* Fix SQL error showing at top of page on storefronts in PHP8 by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/171
* Remove unused lease payment due template from admin by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/173
* Fix PHP error in some PHP versions caused by old constructor naming by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/175
* Fix PHP8 issues with admin template management by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/176
* Change the upload_max_filesize to 64M on the docker php.ini settings by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/177
* Fixing PHP8 issues when browsing categories with core display addon e… by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/178
* Fix PHP8 compatibility issues with advanced search by @jonyo in https://github.com/geodesicsolutions-community/geocore-community/pull/179

## New Contributors
* @jonyo made their first contribution in https://github.com/geodesicsolutions-community/geocore-community/pull/7
* @vicos59 made their first contribution in https://github.com/geodesicsolutions-community/geocore-community/pull/118

**Full Changelog**: https://github.com/geodesicsolutions-community/geocore-community/compare/v0.0.0...v20.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/geodesicsolutions-community/geocore-community/releases/tag/v20.0.0'>v20.0.0 First Stable Release!</a>.</em>

**Comment by ugintl** [comment]
There are 4 zip folders. Should I download all of them? or just geocore-ce.zip?

**Reply by vicos59** [reply]
geocore-ce.zip is all you really need. You might also be interested in one of the optional template sets. Personally, I would start out with the default template and then once everything is configured and working experiment with the other ones.

**Comment by phillic86** [comment]
Hello, there is an error when we are in the admin panel and we want to add a language. Is it normal? Warmly

**Comment by ncrs-admin** [comment]
I upgraded from Ver 18 to Ver 20 and it is running fine on PHP 7.4. When I change to PHP 8 everything works except the link to Contact Seller. There are apparently a lot of Smarty issues in building the contact form. I get about 20 errors. I know nothing about smarty and have tried to fix some of the issues, but no luck.

Anyone working on that issue?
